logstash-core/lib/logstash/modules/kibana_config.rb

```diff
@@ -0,0 +1,104 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging"
+
+require_relative "file_reader"
+require_relative "kibana_resource"
+require_relative "kibana_base_resource"
+
+module LogStash module Modules class KibanaConfig
+  include LogStash::Util::Loggable
+
+  ALLOWED_DIRECTORIES = ["search", "visualization"]
+
+  attr_reader :index_name
+
+  # We name it `modul` here because `module` has meaning in Ruby.
+  def initialize(modul, settings)
+    @directory = ::File.join(modul.directory, "kibana")
+    @name = modul.module_name
+    @settings = settings
+    @index_name = settings.fetch("dashboards.kibana_index", ".kibana")
+  end
+
+  def dashboards
+    # there can be more than one dashboard to load
+    filenames = FileReader.read_json(dynamic("dashboard"))
+    filenames.map do |filename|
+      KibanaResource.new(@index_name, "dashboard", dynamic("dashboard", filename))
+    end
+  end
+
+  def index_pattern
+    pattern_name = "#{@name}-*"
+    default_index_json = '{"defaultIndex": "#{pattern_name}"}'
+    default_index_content_id = @settings.fetch("index_pattern.kibana_version", "5.4.0") # make this 5.5.0
```

Comment:
For each LS release we will have to alter the string or the constant until we switch to the kibana API if it covers index patterns or their enhancement that picks an index-pattern as default. Note: our module solution does not require the setting of a default-index to actually work but the kibana UI bleats and moans until the user "picks" a default index pattern and in doing that the user could end up "editing" the index-pattern we so lovingly imported for them.
The PUT target is `.kibana/config/<current-kibana-version>` - IFAICT we can't query kibana for its version.
So until we can do better for the user they are going to have to tell us that they are running with a different version of kibana than the one we have released the code for i.e `5.5.0`, `5.6.0` or `6.0.0`.
Its hard. If we release the for `6.0.0` and the user later downloads `kibana 6.0.2` to fix a different bug then there will not be a default index in the kibana UI and BOOM - IDK how we successfully message this to the user.